OpenSim/Simulation/Model/PathPoint.h

```diff
@@ -35,88 +35,76 @@ class PhysicalFrame;
 //=============================================================================
 //=============================================================================
 /**
- * A class implementing a path point.
- *
- * @author Peter Loan
- * @version 1.0
+ * A PathPoint that is stationary with respect to parent's PhysicalFrame
  */
-
-//Preserve old PathPoint behavior while basing on concrete Points
-
-class OSIMSIMULATION_API PathPoint : public Station {
-OpenSim_DECLARE_CONCRETE_OBJECT(PathPoint, Station);
+class OSIMSIMULATION_API PathPoint : public AbstractPathPoint {
+    OpenSim_DECLARE_CONCRETE_OBJECT(PathPoint, AbstractPathPoint);
+public:
+//==============================================================================
+// PROPERTIES
+//==============================================================================
+    OpenSim_DECLARE_PROPERTY(location, SimTK::Vec3,
+        "The fixed location of the station expressed in its parent frame.");
 //=============================================================================
 // METHODS
 //=============================================================================
-    //--------------------------------------------------------------------------
-    // CONSTRUCTION
-    //--------------------------------------------------------------------------
-public:
-    PathPoint() : Station() {}
+    PathPoint();
 
-#ifndef SWIG
-    const SimTK::Vec3& getLocation() const { return get_location(); }
-#endif
-    SimTK::Vec3& getLocation()  { return upd_location(); }
-
-    const double& getLocationCoord(int aXYZ) const {
-        assert(aXYZ>=0 && aXYZ<=2); return get_location()[aXYZ];
-    }
-
-    // A variant that uses basic types for use by GUI
-    void setLocationCoord(int aXYZ, double aValue) {
+    // A variant that uses basic types for use by GUI, e.g. OpenSimContext
+    /** <b>(Deprecated)</b> Old PathPoint interface */
+    DEPRECATED_14("Use setLocation() instead.")
+    void setLocationCoord(const SimTK::State&s, int aXYZ, double aValue) {
```

Comment:
Agreed @tkuchida :+1:, this was written before SimTK classes were exposed to Java/Matlab but it looks like this can easily be done now. Would be good to keep track of it but doesn't seem necessary/related to this PR. The OpenSimContext class will likely have to undergo some changes anyway when the MuscleEditor is refactored/rewritten and will be slimmed down significantly.